In [1]:
import pandas as pd
import numpy as np
import plotly.express as px
import plotly.graph_objects as go

# Chargement des données
df = pd.read_csv('../data/hotel_prices.csv', parse_dates=['date'])

print(f"Dataset chargé : {df.shape}")
print(f"\nAperçu :")
print(df.head())

# Visualisation 1 — Prix moyen par ville
prix_ville = df.groupby('ville')['prix_nuit'].mean().round(2).reset_index()
prix_ville = prix_ville.sort_values('prix_nuit', ascending=False)

fig1 = px.bar(
    prix_ville,
    x='ville',
    y='prix_nuit',
    title='Prix moyen par nuit par ville — Côte d\'Azur & Monaco',
    color='ville',
    color_discrete_sequence=px.colors.qualitative.Set2,
    text='prix_nuit'
)
fig1.update_traces(texttemplate='%{text:.0f} €', textposition='outside')
fig1.update_layout(
    yaxis_title="Prix moyen (€)",
    xaxis_title="Ville",
    showlegend=False
)
fig1.show()

Dataset chargé : (1800, 8)

Aperçu :
  ville           hotel  categorie       date jour_semaine  mois  weekend  \
0  Nice  Hotel Negresco  5 étoiles 2024-06-01     Saturday  June     True   
1  Nice  Hotel Negresco  5 étoiles 2024-06-02       Sunday  June     True   
2  Nice  Hotel Negresco  5 étoiles 2024-06-03       Monday  June    False   
3  Nice  Hotel Negresco  5 étoiles 2024-06-04      Tuesday  June    False   
4  Nice  Hotel Negresco  5 étoiles 2024-06-05    Wednesday  June    False   

   prix_nuit  
0     526.45  
1     588.68  
2     470.88  
3     458.88  
4     419.04  


In [2]:
# Visualisation 2 — Évolution des prix dans le temps par ville
prix_temps = df.groupby(['date', 'ville'])['prix_nuit'].mean().round(2).reset_index()

fig2 = px.line(
    prix_temps,
    x='date',
    y='prix_nuit',
    color='ville',
    title='Évolution des prix sur 60 jours — Côte d\'Azur & Monaco',
    color_discrete_sequence=px.colors.qualitative.Set2
)
fig2.update_layout(
    yaxis_title="Prix moyen (€)",
    xaxis_title="Date",
    legend_title="Ville"
)
fig2.show()

In [3]:
# Visualisation 3 — Weekend vs Semaine par ville
prix_weekend = df.groupby(['ville', 'weekend'])['prix_nuit'].mean().round(2).reset_index()
prix_weekend['periode'] = prix_weekend['weekend'].map({True: 'Weekend', False: 'Semaine'})

fig3 = px.bar(
    prix_weekend,
    x='ville',
    y='prix_nuit',
    color='periode',
    barmode='group',
    title='Prix moyen Weekend vs Semaine par ville',
    color_discrete_map={'Weekend': '#E74C3C', 'Semaine': '#3498DB'},
    text='prix_nuit'
)
fig3.update_traces(texttemplate='%{text:.0f} €', textposition='outside')
fig3.update_layout(
    yaxis_title="Prix moyen (€)",
    xaxis_title="Ville",
    legend_title="Période"
)
fig3.show()

In [4]:
# Visualisation 4 — Prix par catégorie et ville
prix_cat = df.groupby(['ville', 'categorie'])['prix_nuit'].mean().round(2).reset_index()

fig4 = px.bar(
    prix_cat,
    x='categorie',
    y='prix_nuit',
    color='ville',
    barmode='group',
    title='Prix moyen par catégorie d\'hôtel et par ville',
    color_discrete_sequence=px.colors.qualitative.Set2,
    text='prix_nuit'
)
fig4.update_traces(texttemplate='%{text:.0f} €', textposition='outside')
fig4.update_layout(
    yaxis_title="Prix moyen (€)",
    xaxis_title="Catégorie",
    legend_title="Ville"
)
fig4.show()

## Conclusions & insights business

### 🏆 Classement des villes par prix moyen
1. Monaco — 860 € (+163% vs Menton)
2. Antibes — 526 €
3. Cannes — 501 €
4. Nice — 428 €
5. Juan-les-Pins — 373 €
6. Menton — 326 €

### 📅 Impact saisonnalité
- Weekend vs Semaine : **+35 à 38%** dans toutes les villes
- Grand Prix F1 Monaco (juin) : **+250%** sur les prix
- Fête Nationale (14 juillet) : **+60%** en moyenne
- Haute saison juillet : **+30%** vs juin

### 💡 Recommandations pour les hôteliers
- Un hôtel niçois qui ne pratique pas de tarif weekend perd ~38% de revenus potentiels
- La période Grand Prix est la plus lucrative de l'année — anticiper 3-4 mois à l'avance
- Menton est sous-valorisé — potentiel de hausse de prix inexploité